src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.component.js

```diff
@@ -1,65 +1,50 @@
 import React from "react";
 import PropTypes from "prop-types";
 import StyledToolbarButton from "./toolbar-button.style";
-import TooltipDecorator from "../../../../../utils/decorators/tooltip-decorator/tooltip-decorator";
 
-class ToolbarButton extends React.Component {
-  componentDidUpdate() {
-    // ensure tooltip is hidden when prop updates
-    if (!this.props.tooltipVisible) {
-      this.onHide();
-    }
-  }
-
-  render() {
+const ToolbarButton = React.forwardRef(
+  (
+    {
+      onKeyDown,
+      onMouseDown,
+      activated,
+      ariaLabel,
+      tabbable,
+      children,
+      ...rest
+    },
+    ref
+  ) => {
     return (
-      <>
-        <StyledToolbarButton
-          data-component="text-editor-toolbar-button"
-          ref={(comp) => {
-            this._target = comp;
-          }}
-          onMouseOver={this.props.onMouseOver}
-          onMouseLeave={this.props.onMouseLeave}
-          onFocus={this.props.onMouseOver}
-          onKeyDown={this.props.onKeyDown}
-          onMouseDown={this.props.onMouseDown}
-          onBlur={this.props.onMouseLeave}
-          isActive={this.props.activated}
-          aria-label={this.props.ariaLabel}
-          {...(!this.props.tabbable && { tabIndex: -1 })}
-        >
-          {this.props.children}
-        </StyledToolbarButton>
-
-        {this.tooltipHTML}
-      </>
+      <StyledToolbarButton
+        data-component="text-editor-toolbar-button"
+        ref={ref}
+        onKeyDown={onKeyDown}
+        onMouseDown={onMouseDown}
+        isActive={activated}
+        aria-label={ariaLabel}
+        {...(!tabbable && { tabIndex: -1 })}
+        {...rest}
```

Comment:
I don't think we just want to be passing down all of the props to the styled component here, only ones we specifically need

Reply:
It's an internal component that's only composed within the TextEditor, but I'll update it for the sake of maintainability